pcsx2-qt/ShortcutCreationDialog.cpp

```diff
@@ -0,0 +1,441 @@
+	});
+
+	connect(m_ui.overrideBootELFToggle, &QCheckBox::toggled, m_ui.overrideBootELFPath, &QLineEdit::setEnabled);
+	connect(m_ui.overrideBootELFToggle, &QCheckBox::toggled, m_ui.overrideBootELFButton, &QPushButton::setEnabled);
+	connect(m_ui.gameArgsToggle, &QCheckBox::toggled, m_ui.gameArgs, &QLineEdit::setEnabled);
+	connect(m_ui.loadStateIndexToggle, &QCheckBox::toggled, m_ui.loadStateIndex, &QSpinBox::setEnabled);
+	connect(m_ui.loadStateFileToggle, &QCheckBox::toggled, m_ui.loadStateFilePath, &QLineEdit::setEnabled);
+	connect(m_ui.loadStateFileToggle, &QCheckBox::toggled, m_ui.loadStateFileBrowse, &QPushButton::setEnabled);
+	connect(m_ui.bootOptionToggle, &QCheckBox::toggled, m_ui.bootOptionDropdown, &QPushButton::setEnabled);
+	connect(m_ui.fullscreenMode, &QCheckBox::toggled, m_ui.fullscreenModeDropdown, &QPushButton::setEnabled);
+
+	m_ui.shortcutDesktop->setChecked(true);
+	m_ui.overrideBootELFPath->setEnabled(false);
+	m_ui.overrideBootELFButton->setEnabled(false);
+	m_ui.gameArgs->setEnabled(false);
+	m_ui.bootOptionDropdown->setEnabled(false);
+	m_ui.fullscreenModeDropdown->setEnabled(false);
+	m_ui.loadStateIndex->setEnabled(false);
+	m_ui.loadStateFileBrowse->setEnabled(false);
+	m_ui.loadStateFilePath->setEnabled(false);
+
+	m_ui.customArgsInstruction->setText(tr("You may add additional (space-separated) <a href=\"https://pcsx2.net/docs/post/cli/\">custom arguments</a> that are not listed above here:"));
+	m_ui.customArgsInstruction->setTextFormat(Qt::RichText);
+	m_ui.customArgsInstruction->setTextInteractionFlags(Qt::TextBrowserInteraction);
+	m_ui.customArgsInstruction->setOpenExternalLinks(true);
+
+	if (std::getenv("container"))
+	{
+		m_ui.portableModeToggle->setEnabled(false);
+		m_ui.shortcutDesktop->setEnabled(false);
+		m_ui.shortcutStartMenu->setEnabled(false);
+	}
+
+	connect(m_ui.dialogButtons, &QDialogButtonBox::accepted, this, [&]() {
+		std::vector<std::string> args;
+
+		if (m_ui.portableModeToggle->isChecked())
+			args.push_back("-portable");
+
+		if (m_ui.overrideBootELFToggle->isChecked() && !m_ui.overrideBootELFPath->text().isEmpty())
+		{
+			args.push_back("-elf");
+			args.push_back(m_ui.overrideBootELFPath->text().toStdString());
+		}
+
+		if (m_ui.gameArgsToggle->isChecked() && !m_ui.gameArgs->text().isEmpty())
+		{
+			args.push_back("-gameargs");
+			args.push_back(m_ui.gameArgs->text().toStdString());
+		}
+
+		if (m_ui.bootOptionToggle->isChecked() && m_ui.bootOptionDropdown->currentIndex() == 0)
+			args.push_back("-fastboot");
+		else if (m_ui.bootOptionToggle->isChecked() && m_ui.bootOptionDropdown->currentIndex() == 1)
+			args.push_back("-slowboot");
+
+		if (m_ui.loadStateIndexToggle->isChecked() && (m_ui.loadStateIndex->value() >= 1 && m_ui.loadStateIndex->value() <= 10))
+		{
+			args.push_back("-state");
+			args.push_back(StringUtil::ToChars(m_ui.loadStateIndex->value()));
+		}
+		else if (m_ui.loadStateIndexToggle->isChecked() && !m_ui.loadStateFilePath->text().isEmpty())
```

Comment:
Loading a save state from a file doesn't work since the condition here is wrong.